src/som/vmobjects/SClass.java

```diff
@@ -56,6 +60,7 @@
   @CompilationFinal private HashSet<SlotDefinition> slots; // includes slots of super classes and mixins
 
   @CompilationFinal private MixinDefinition mixinDef;
+  @CompilationFinal private VirtualFrame context; // a frame containing the outer scope (implicit classes only)
```

Comment:
Please check the [`context`](https://github.com/smarr/SOMns/blob/master/src/som/vmobjects/SBlock.java#L38) field in blocks (and the nodes that create block objects).
This frame needs to be *materialized*.